.travis.yml

```diff
@@ -53,21 +46,95 @@ install:
     - cpanm --quiet --installdeps --notest .
     - cpanm --quiet --notest Devel::Cover::Report::Coveralls
 
+# Setup the MySQL server
 before_script:
     - mysql -u root -h localhost -e 'GRANT ALL PRIVILEGES ON *.* TO "travis"@"%"'
 
-script: travis_wait 40 "./travisci/harness.sh"
 
+# What jobs to run and in which order
 jobs:
   include:
-  - stage: trigger_dependent_builds
-    script: "./travisci/trigger-dependent-build.sh"
-
-# Get the matrix to only build coveralls support on the latest Perl
-matrix:
-  exclude:
-    - perl: "5.14"
-      env: COVERALLS=true
-    - perl: "5.30"
-      env: COVERALLS=false
+    - language: perl
+      perl: 5.30
+      name: "Housekeeping (all languages)"
+      install:
+        - cpanm --quiet --installdeps --notest --cpanfile ensembl/cpanfile .
+      before_script:
+        - echo
+      script: ./travisci/all-housekeeping_harness.sh
+
+    - language: perl
+      dist: trusty   # 5.22 is the minimum on xenial
+      perl: 5.14
+      name: "Perl unit tests on the minimum version"
+      env:
+        - COVERAGE=false
+      script: ./travisci/perl-unittest_harness.sh
+
+    - language: perl
+      perl: 5.30
+      name: "Perl unit tests on the latest version, with code coverage"
+      env:
+        - COVERAGE=true
+      script: ./travisci/perl-unittest_harness.sh
+
+    - language: perl
+      perl: 5.30
+      name: "Perl linter"
+      install:
+        - cpanm --quiet --installdeps --notest --cpanfile ensembl/cpanfile .
+      before_script:
+        - echo
+      script: ./travisci/perl-linter_harness.sh
+
+    - language: python
+      python: 3.4
```

Comment:
Should be 3.5?

Reply:
3.4 there was the minimum I could get pytest to run on (I've tried the previous ones). But yes, 3.4 not being supported any more, it should be 3.5